lib/charms/mysql/v0/mysql.py

```diff
@@ -2154,101 +1807,66 @@ def rejoin_instance_to_cluster(
         ):
             raise MySQLLockAcquisitionError("Lock not acquired")
 
         try:
-            logger.debug(f"Rejoining instance {unit_address} to cluster {self.cluster_name}")
-            self._run_mysqlsh_script(
-                "\n".join(commands),
-                user=self.server_config_user,
-                password=self.server_config_password,
-                host=self.instance_def(self.server_config_user, from_instance),
-            )
-        except MySQLClientError as e:
-            logger.error(
-                f"Failed to rejoin instance {unit_address} to cluster {self.cluster_name}"
+            # NOTE: This operation used to be done by the server-config user
+            client.rejoin_instance_into_cluster(
+                cluster_name=self.cluster_name,
+                instance_host=unit_address,
+                instance_port=str(3306),
+                options=options,
             )
-            raise MySQLRejoinInstanceToClusterError from e
+        except ExecutionError as e:
+            raise MySQLRejoinInstanceToClusterError() from e
         finally:
-            # always release the lock
-            self._release_lock(from_instance, unit_label, UNIT_ADD_LOCKNAME)
+            self._release_lock(
+                executor=executor,
+                unit_label=unit_label,
+                unit_task=CharmLockingQueryBuilder.INSTANCE_ADDITION_TASK,
+            )
 
-    def is_instance_configured_for_innodb(
-        self, instance_address: str, instance_unit_label: str
-    ) -> bool:
+    def is_instance_configured_for_innodb(self, instance_address: str) -> bool:
         """Confirm if instance is configured for use in an InnoDB cluster."""
-        commands = (
-            "instance_configured = dba.check_instance_configuration()['status'] == 'ok'",
-            'print("INSTANCE_CONFIGURED" if instance_configured else "INSTANCE_NOT_CONFIGURED")',
+        client = MySQLClusterClient(
+            executor=self._build_instance_tcp_executor(instance_address),
         )
 
         try:
-            logger.debug(
-                f"Confirming instance {instance_address}/{instance_unit_label} configuration for InnoDB"
-            )
-
-            output = self._run_mysqlsh_script(
-                "\n".join(commands),
-                user=self.server_config_user,
-                password=self.server_config_password,
-                host=self.instance_def(self.server_config_user, instance_address),
-            )
-            return "INSTANCE_CONFIGURED" in output
-        except MySQLClientError as e:
-            # confirmation can fail if the clusteradmin user does not yet exist on the instance
-            logger.warning(
-                f"Failed to confirm instance configuration for {instance_address} with error {e.message}",
-            )
+            result = client.check_instance_before_cluster()
+        except ExecutionError:
```

Comment:
```suggestion
        except ExecutionError as exc:
            logger.debug(exc)
```

Wondering if we should at least do this, here and in other places. Otherwise these masked exceptions might be difficult to debug.